native/native_image.go

```diff
@@ -68,26 +70,28 @@ func (n NativeImage) Contribute(layer libcnb.Layer) (libcnb.Layer, error) {
 		return libcnb.Layer{}, fmt.Errorf("unable to process arguments\n%w", err)
 	}
 
+	buf := &bytes.Buffer{}
+	if err := n.Executor.Execute(effect.Execution{
+		Command: "native-image",
+		Args:    []string{"--version"},
+		Stdout:  buf,
+		Stderr:  n.Logger.BodyWriter(),
+	}); err != nil {
+		return libcnb.Layer{}, fmt.Errorf("error running version\n%w", err)
+	}
+	nativeBinaryHash := fmt.Sprintf("%x", sha256.Sum256(buf.Bytes()))
+
 	contributor := libpak.NewLayerContributor("Native Image", map[string]interface{}{
 		"files":       files,
 		"arguments":   arguments,
 		"compression": n.Compressor,
+		"info":        nativeBinaryHash,
```

Comment:
```suggestion
		"version-hash":        nativeBinaryHash,
```

Very minor, but just trying to be a little more descriptive. Your call.